examples/ExactlyOnceBasic/Program.cs

```diff
@@ -0,0 +1,680 @@
+            await CreateTopic(adminClient, reversedWordsTopicName, reversedWordsTopicPartitions);
+
+            producerConfig.TransactionalId = transactionalId;
+            var consumerBuilder = new ConsumerBuilder<Null, string>(consumerConfig);
+            var producerBuilder = new ProducerBuilder<Null, string>(producerConfig);
+            var retryInterval = generalConfig.GetValue<TimeSpan>("RetryInterval");
+            var localTransactionOperationTimeout = generalConfig.GetValue<TimeSpan>("LocalTransactionOperationTimeout");
+
+            switch (command)
+            {
+                case "create_words":
+                    new WordsCreator
+                    {
+                        OutputTopic = wordsTopicName,
+                        ProduceRate = wordsConfig.GetValue<TimeSpan>("ProduceRate"),
+                        RetryInterval = retryInterval,
+                        LocalTransactionOperationTimeout = localTransactionOperationTimeout,
+                        ProducerBuilder = producerBuilder,
+                        Log = Program.Log,
+                        CTS = cancelled
+                    }
+                    .Run();
+                    break;
+                case "reverse_words":
+                    new WordReverser
+                    {
+                        InputTopic = wordsTopicName,
+                        OutputTopic = reversedWordsTopicName,
+                        CommitMaxMessages = reversedWordsConfig.GetValue<int>("CommitMaxMessages"),
+                        CommitPeriod = reversedWordsConfig.GetValue<TimeSpan>("CommitPeriod"),
+                        CommitTimeout = reversedWordsConfig.GetValue<TimeSpan>("CommitTimeout"),
+                        RetryInterval = retryInterval,
+                        LocalTransactionOperationTimeout = localTransactionOperationTimeout,
+                        ProducerBuilder = producerBuilder,
+                        ConsumerBuilder = consumerBuilder,
+                        Log = Program.Log,
+                        CTS = cancelled
+                    }
+                    .Run();
+                    break;
+                default:
+                    Log("Unknown command");
+                    break;
+            }
+            completed.Cancel();
+        }
+    }
+
+    /// <summary>
+    ///     This transactional producer allows to use a producer an consumer for transactions,
+    ///     by recreating the producer when a fatal error happens, sending offsets to transaction
+    ///     and handling aborts and retries correctly.
+    /// </summary>
+    public class TransactionalProducer<K, V> : IDisposable
+    {
+        public string OutputTopic { get; set; }
+
+        public Action<string> Log { get; set; }
+
+        public TimeSpan RetryInterval { get; set; }
+
+        public TimeSpan LocalTransactionOperationTimeout { get; set; }
+
+        public ProducerBuilder<K, V> ProducerBuilder { get; set; }
```

Comment:
I wouldn't maintain a builder class like this - create a builder class every time you need to create the producer is how i intended. it is lightweight.